internal/saucecloud/zip/archive.go

```diff
@@ -0,0 +1,138 @@
+package zip
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/saucelabs/saucectl/internal/archive/zip"
+	"github.com/saucelabs/saucectl/internal/jsonio"
+	"github.com/saucelabs/saucectl/internal/msg"
+	"github.com/saucelabs/saucectl/internal/node"
+	"github.com/saucelabs/saucectl/internal/sauceignore"
+)
+
+// ArchiveFileCountSoftLimit is the threshold count of files added to the archive
+// before a warning is printed.
+// The value here (2^15) is somewhat arbitrary. In testing, ~32K files in the archive
+// resulted in about 30s for download and extraction.
+const ArchiveFileCountSoftLimit = 32768
+
+func ArchiveRunnerConfig(project interface{}, tempDir string) (string, error) {
```

Comment:
Now that you have extracted these privates methods and exported them, they do need documentation.